### DISCUSSION TITLE
Integration with MinIO or explore options

### DISCUSSION THREAD

**Discussion by saveriogzz** [slskd/slskd#1060] [Q&A]
Hello! First of all thank you so much to the developers and maintainers for this great piece of software, it's really something!

I'm planning on deploying slskd on a RaspberryPi, which is directly connected to an hard disk that is used as a [minio](https://github.com/minio/minio) volume, whose instance is also running on the same Raspberry. Minio is basically an open source AWS S3 and it's great!

A way (perhaps not the only one) to upload data to a minio instance is to use the minio client - [mc](https://github.com/minio/mc) - which also has a [dotNET SDK](https://github.com/minio/minio-dotnet). This kind of setup with slskd+minio could be easily deployed and managed with something like [balena](https://www.balena.io/) but this will be a second step.

I'd like to ask the community's feedback of such a setup. Do you find it acceptable? How would you do it otherwise? Would an integration with MinIO fit nicely within slskd's roadmap?

I have zero C# experience but know Scala thanks to my data engineering skills. However I'm not sure how transferrable this is, I'm always up for learning something new!

Thanks in advance!

**Comment by jpdillingham** [comment]
I built the [Relay](https://github.com/slskd/slskd/blob/master/docs/relay.md) feature to allow for distributed scenarios; I think that might interest you.

slskd needs the ability to scan the filesystem where files are stored, and this type of I/O does not work great over a network, but network paths are supported.

Is it possible to mount a minio volume like a normal disk?  If so it should be simple to map a share to it.
